integration/dataproc/build.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+# Running this script will create edited versions of the files in the .generated directory
+set -e
+
+ALLUXIO_DOWNLOAD_URL=${1}
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+mkdir -p ${DIR}/.generated
+cp ${DIR}/alluxio.sh ${DIR}/.generated
+
+# replace ALLUXIO_DOWNLOAD_URL in dataproc init script (alluxio.sh)
+if [[ -n ${ALLUXIO_DOWNLOAD_URL} ]]; then
+    if [[ -n $(grep "readonly ALLUXIO_DOWNLOAD_URL=\"${ALLUXIO_DOWNLOAD_URL}\"" "${DIR}/alluxio.sh") ]]; then
```

Comment:
this doesn't account for the "fail to find the string to replace" scenario
if i were refactoring the script and renamed ALLUXIO_DOWNLOAD_URL to ALLUXIO_DOWNLOAD_S3_URL, this script would not replace anything but pass.
```
if alluxio download url is set {
  if string to replace isn't found by grep -> fail
  replace string with perl cmd
}
```
it's fine to replace the same string; that's not much of a concern. i brought up the scenario as a counterexample to why using `diff` wouldn't be as dependable as using `grep`